tests/test_errors.py

```diff
@@ -7,24 +7,38 @@
 
 
 def test_exception_for_response_with_description():
+    """
+    When we catch an exception with a populated Response object,
+    the error message should extract data from the object
+    and format the message. Then we assert it's as expected
+    with an f-string pattern.
```

Comment:
nit: imo "Then we assert it's as expected with an f-string pattern" is a test implementation detail and doesn't provide relevant information to the reader about what the test does or why and thus could be removed. Same comment for other tests.

Reply:
You mean opposed to the nothing that was there before?

Reply:
Obviously this is infinitely better than no docstring; I never argued otherwise. I just think this specific part doesn't bring much